src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -72,23 +75,54 @@ public void inferenceSentences(
         @NonNull final List<String> inputText,
         @NonNull final ActionListener<List<List<Float>>> listener
     ) {
-        final ModelResultFilter modelResultFilter = new ModelResultFilter(false, true, targetResponseFilters, null);
-        final MLInputDataset inputDataset = new TextDocsInputDataSet(inputText, modelResultFilter);
-        final MLInput mlInput = new MLInput(FunctionName.TEXT_EMBEDDING, null, inputDataset, MLModelTaskType.TEXT_EMBEDDING);
-        final List<List<Float>> vector = new ArrayList<>();
-
+        MLInput mlInput = createMLInput(targetResponseFilters, inputText);
         mlClient.predict(modelId, mlInput, ActionListener.wrap(mlOutput -> {
-            final ModelTensorOutput modelTensorOutput = (ModelTensorOutput) mlOutput;
-            final List<ModelTensors> tensorOutputList = modelTensorOutput.getMlModelOutputs();
-            for (final ModelTensors tensors : tensorOutputList) {
-                final List<ModelTensor> tensorsList = tensors.getMlModelTensors();
-                for (final ModelTensor tensor : tensorsList) {
-                    vector.add(Arrays.stream(tensor.getData()).map(value -> (Float) value).collect(Collectors.toList()));
-                }
-            }
+            final List<List<Float>> vector = buildVectorFromResponse(mlOutput);
             log.debug("Inference Response for input sentence {} is : {} ", inputText, vector);
             listener.onResponse(vector);
         }, listener::onFailure));
     }
 
+    /**
+     * Abstraction to call predict function of api of MLClient with provided targetResponseFilters. It uses the
+     * custom model provided as modelId and run the {@link MLModelTaskType#TEXT_EMBEDDING}. The return will be sent
+     * using the actionListener which will have a {@link List} of {@link List} of {@link Float} in the order of
+     * inputText. We are not making this function generic enough to take any function or TaskType as currently we need
+     * to run only TextEmbedding tasks only. Please note this method is a blocking method, use this only when the processing
+     * needs block waiting for response, otherwise please use {@link #inferenceSentences(String, List, ActionListener)}
+     * instead.
+     * @param modelId {@link String}
+     * @param inputText {@link List<String>} on which inference needs to happen.
+     * @return {@link List} of {@link List<String>} represents the text embedding vector result.
```

Comment:
```suggestion
     * @param inputText {@link List} of {@link String} on which inference needs to happen.
     * @return {@link List} of {@link List} of {@link String} represents the text embedding vector result.
```

Do this, otherwise it is a java doc error.